### PR TITLE
Fix linting and bad copying in api package

### PR DESCRIPTION
**Pull request by chases2**
"bad copying" == "Don't copy locks" and "Don't copy big protos"

/assign @fejta 
/lint

### CODE REVIEW

File: pkg/api/v1/config.go [hunk 1]
```diff
@@ -14,6 +14,7 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
+// Package api.v1 is the first versioned implementation of the API
```
[the remark this answers is not in the document]
Comment (chases2) [reply pkg/api/v1/config.go]: Fixed

File: pkg/api/v1/config_test.go [hunk 2]
```diff
@@ -154,21 +155,22 @@ func TestListDashboardGroups(t *testing.T) {
...
-			t.Errorf("Expected OK status, but got %v", response.Code)
-		}
-		if response.Body.String() != test.expectedJSON {
-			t.Errorf("In Body, Expected %s; got %s", test.expectedJSON, response.Body.String())
-		}
+		t.Run(test.name, func(t *testing.T) {
+			server := setupTestServer(t, test.config)
+			request, err := http.NewRequest("GET", "/api/v1/dashboardgroups", nil)
+			if err != nil {
+				t.Fatalf("Can't form dashboardgroup request")
```
Comment (fejta) [inline pkg/api/v1/config_test.go:162]: Include the error message?

`t.Fatalf("whatever: %v", err)`

File: pkg/api/v1/config_test.go [hunk 3]
```diff
@@ -154,21 +155,22 @@ func TestListDashboardGroups(t *testing.T) {
...
+			}
+			response := httptest.NewRecorder()
+			server.ListDashboardGroups(response, request)
+			if response.Code != http.StatusOK {
```
Comment (fejta) [inline pkg/api/v1/config_test.go:166]: Shouldn't we also test non-happy paths?

**Review by fejta — APPROVED** [review]
/hold

File: pkg/api/v1/config_test.go (hunk 2 above)
Comment (chases2) [reply pkg/api/v1/config_test.go]: Done.

File: pkg/api/v1/config_test.go (hunk 3 above)
Comment (chases2) [reply pkg/api/v1/config_test.go]: Done. Expanded this test to include passed-in scopes and trying to read an invalid config.

**Pull request merged**